api/src/main/java/org/openmrs/module/queue/api/QueueEntryService.java

```diff
@@ -85,36 +96,42 @@ public interface QueueEntryService {
 	 * @param queueEntry queue entry to be deleted
 	 * @throws org.openmrs.api.APIException
 	 */
+	@Authorized({ PrivilegeConstants.PURGE_QUEUE_ENTRIES })
 	void purgeQueueEntry(@NotNull QueueEntry queueEntry) throws APIException;
 	
 	/**
 	 * @return {@link List} of queue entries that match the given %{@link QueueEntrySearchCriteria}
 	 */
+	@Authorized({ PrivilegeConstants.GET_QUEUE_ENTRIES })
 	List<QueueEntry> getQueueEntries(@NotNull QueueEntrySearchCriteria searchCriteria);
 	
 	/**
 	 * @return {@link Long} count of queue entries that match the given
 	 *         %{@link QueueEntrySearchCriteria}
 	 */
+	@Authorized({ PrivilegeConstants.GET_QUEUE_ENTRIES })
 	Long getCountOfQueueEntries(@NotNull QueueEntrySearchCriteria searchCriteria);
 	
 	/**
 	 * @param location
 	 * @param queue
 	 * @return VisitQueueNumber - used to identify patients in the queue instead of using patient name
 	 */
+	@Authorized({ ADD_VISITS, EDIT_VISITS })
```

Comment:
Why not use the manage privilege for the item where the visit queue number is used?

Reply:
After looking at the implementation details of this method, i think your privileges are correct. 😊

Reply:
Instead of import static, how about directly using `org.openmrs.util.PrivilegeConstants.ADD_VISITS` and `org.openmrs.util.PrivilegeConstants.EDIT_VISITS`